### DISCUSSION TITLE
Should `app.kubernetes.io/managed-by: Helm` be removed for Helm charts output?

### DISCUSSION THREAD

**Discussion by iliapolo** [cdk8s-team/cdk8s-core#746] [General]
*Original Author*: [akefirad](https://github.com/akefirad)

----------------------------

I'm not deeply familiar how Helm works and not sure if it's really needed. But it seems to me that having the fabled at the very least is confusing, since the app is not really managed by Helm, in my case it's managed via ArgoCD. Does that make sense to remove the label from the generated templates? Currently I'm doing it using a JsonPatch.remove. Any thoughts? Thanks.

**Comment by iliapolo** [comment]
@akefirad Can you share an example of what exactly do you mean? Are you suggesting CDK8s remove helm related label when it [imports helm templates](https://cdk8s.io/docs/latest/basics/helm/)?

**Comment by akefirad** [comment]
Right, so my understanding is CDK8S uses Helm only to generate the template, but the deployed resources are not managed by Helm. Now when Helm generates manifests, (I think) it adds a bunch of labels, including `app.kubernetes.io/managed-by: Helm` which may or may not be significant. Now my question is, if the label is significant and confused Helm (if anyone uses both CDK8S and Helm together), should we remove the label in the generated manifest?

**Comment by akefirad** [comment]
Coming back to this, looking at the [recommendations](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-managed-by), having this label can (potentially) cause confusion for people and tools.

**Comment by thiagowfx** [comment]
It looks like this: <img width="1556" alt="image" src="https://github.com/user-attachments/assets/6cfa57cb-7e28-4ec1-85b1-f5a6251ba9eb" />

[Related](https://perrotta.dev/2025/06/argocd-upgrade-to-v3-remove-helm-labels/).

**Comment by akefirad** [comment]
World have been nice to do it in CDK8S as well. Currently we doing it using a JSON patch.
